repo2docker/app.py

```diff
@@ -176,6 +176,9 @@ def fetch(self, url, ref, checkout_path):
             sys.exit(1)
 
         if ref:
+            if len(ref.split('/')) != 2:
+                raise ValueError('Expected --ref to be of the form '
+                                 'remote/reference, but got %s' % ref)
```

Comment:
`repo2docker --ref e287c39f62f290e3b6927da08c81586f647d5a4c ...` is currently valid, I think it should continue to be valid. This change would make it invalid though or is the explicit commit hash case handled elsewhere?